sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/PartitionProcessorEventsManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Azure.Messaging.EventHubs;
+using Microsoft.Azure.WebJobs.EventHubs.Processor;
+using Azure.Core;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.Listeners
+{
+    internal class PartitionProcessorEventsManager : IDisposable
+    {
+        private int _maxBatchSize;
+        private int _minBatchSize;
+        private readonly object _cachedEventsLock = new object();
+
+        // This is internal for mocking purposes only.
+        internal Queue<EventData> CachedEvents { get; set; }
+
+        public bool HasCachedEvents
+        {
+            get
+            {
+                lock (_cachedEventsLock)
+                {
+                    return CachedEvents.Count > 0;
+                }
+            }
+        }
+
+        public PartitionProcessorEventsManager(int maxBatchSize, int minBatchSize)
+        {
+            CachedEvents = new Queue<EventData>();
+            _maxBatchSize = maxBatchSize;
+            _minBatchSize = minBatchSize;
+        }
+
+        public void PartitionClosing()
+        {
+            CachedEvents.Clear();
+        }
+
+        public EventData[] GetBatchofEventsWithCached(EventData[] events = null, bool timerTrigger = false)
```

Comment:
```suggestion
        public EventData[] GetBatchofEventsWithCached(EventData[] events = null, bool allowPartialBatch= false)
```

The listener is managing the timer; there's no need for this class to be aware that one is used.  What we're really saying here is whether or not the caller would like a partial batch to be returned.